### PR TITLE
fix: implement `docker` calls as `make` targets

### PR DESCRIPTION
**Pull request by lemeurherve**
This PR replaces `docker` calls by `make` ones so there isn't an unjustified mix of these commands calls anymore and so that everything can be done via `make` commands from CI or locally. (Doc added)

Fixes:
- #420

Extracted from:
- #415

### Testing done

CI

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -44,10 +44,7 @@ pipeline {
                                     environment name: 'IMAGE_TYPE', value: 'linux'
                                 }
                                 steps {
-                                    sh '''
-                                    docker buildx create --use
-                                    docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
-                                    '''
+                                    sh 'make ci-init'
```
Comment (dduportal) [inline Jenkinsfile:47]:
```suggestion
                                    sh 'make buildx-init'
```

nit: the intent is to ensure buildx is initialized, so proposal to (slightly) change the target. The keyword init is good and clear 👍

File: Makefile [hunk 2]
```diff
@@ -37,15 +38,23 @@ check-reqs:
 	@$(call check_cli,curl)
 	@$(call check_cli,jq)
 
+## This function is specific to Jenkins infrastructure and isn't required in other contexts
+ci-init: check-reqs
```
Comment (dduportal) [inline Makefile:42]:
```suggestion
buildx-init: check-reqs
```

File: README.md [hunk 3]
```diff
@@ -196,6 +205,10 @@ make bats
 make: 'bats' is up to date.
 ```
 
+`publish` allows the publication of all images targeted by 'linux' to a registry.
+
+`ci-init` is dedicated to Jenkins infrastructure for initialising docker and isn't required in other contexts.
```
Comment (dduportal) [inline README.md:210]:
```suggestion
`buildx-init` is dedicated to Jenkins infrastructure for initializing docker and isn't required in other contexts.
```

Comment (dduportal) [reply README.md:210]: or

```suggestion
`ci-init` is dedicated to Jenkins infrastructure for initializing docker and isn't required in other contexts.
```

File: Jenkinsfile [hunk 4]
```diff
@@ -61,7 +58,7 @@ pipeline {
                                             sh 'make build'
                                             sh 'make test'
                                             // If the tests are passing for Linux AMD64, then we can build all the CPU architectures
-                                            sh 'docker buildx bake --file docker-bake.hcl linux'
+                                            sh 'make buildall'
```
Comment (dduportal) [inline Jenkinsfile:61]:
```suggestion
                                            sh 'make build-all'
```

nit: use a dashed case as word separator to be homogeneous with the ci-init or check-reqs targets

File: Makefile [hunk 5]
```diff
@@ -37,15 +38,23 @@ check-reqs:
...
 build: check-reqs
-	@set -x; $(bake_base_cli) --set '*.platform=linux/$(ARCH)' $(shell make --silent list)
+	@set -x; $(bake_cli) $(shell make --silent list) --set '*.platform=linux/$(ARCH)'
 
 build-%:
 	@$(call check_image,$*)
-	@set -x; $(bake_base_cli) --set '*.platform=linux/$(ARCH)' '$*'
+	@set -x; $(bake_cli) '$*' --set '*.platform=linux/$(ARCH)'
+
+buildall: check-reqs
```
Comment (dduportal) [inline Makefile:53]:
```suggestion
build-all: check-reqs
```

File: README.md [hunk 6]
```diff
@@ -148,10 +149,18 @@ If you want to test a specific image, you can run:
...
+make test-alpine_jdk17
+```
+
+#### Building all images
+
+You can build all images (even those unsupported by your current architecture) by running:
 
 ```bash
-make test-alpine_jdk11
+make buildall
```
Comment (dduportal) [inline README.md:163]:
```suggestion
make build-all
```

**Review by dduportal — DISMISSED** [review]
LGTM, with a few nits thanks!

File: README.md (hunk 3 above)
Comment (lemeurherve) [reply README.md:210]: Cf comment on #421,

> I chose the English spelling: https://sapling.ai/usage/initialize-vs-initialise, is that OK?

File: Jenkinsfile (hunk 1 above)
Comment (lemeurherve) [reply Jenkinsfile:47]: Cf https://github.com/jenkinsci/docker-ssh-agent/pull/415#discussion_r1644102822

> WDYT of this name as it also setup `docker run --rm --privileged multiarch/qemu-user-static --reset -p yes` (so not only buildx)?

Comment (dduportal) [reply Jenkinsfile:47]: LGTM for `docker-init` !

File: README.md (hunk 3 above)
Comment (dduportal) [reply README.md:210]: No problem for me, *but* I was reminded by @MarkEWaite that we prefer US English by convention in https://github.com/jenkins-infra/jenkins.io/pull/7358#discussion_r1635282081.
I should have added this link earlier, my bad

Comment (lemeurherve) [reply README.md:210]: Good to know, thanks for the link!

File: Jenkinsfile (hunk 1 above)
Comment (lemeurherve) [reply Jenkinsfile]: Addressed in https://github.com/jenkinsci/docker-ssh-agent/pull/421/commits/93f56c49632fa0fa740009bf21390a6d95f76fd5

File: Makefile (hunk 5 above)
Comment (lemeurherve) [reply Makefile:53]: We can't use `build-all` unfortunatelly as there is another make target `build-%` just above already catching every call to `make build-`*.

File: Jenkinsfile (hunk 4 above)
Comment (lemeurherve) [reply Jenkinsfile:61]: We can't use `build-all` unfortunately as there is another make target `build-%` just above already catching every call to `make build-`*.

Comment (lemeurherve) [reply Jenkinsfile:61]: WDYT of `complete-build`?

Comment (lemeurherve) [reply Jenkinsfile:61]: Or "everything"? `make everything`

Comment (lemeurherve) [reply Jenkinsfile:61]: Or `every-build`?

Comment (lemeurherve) [reply Jenkinsfile:61]: After discussing it together, `every-build` seems the best, avoiding the issue of the "build-" prefix already taken and keeping the intent clear.

**Comment by lemeurherve** [comment]
Self-merging as the pull request was approved and the nits have been discussed and resolved.

**Pull request merged**